cmake/Date.cmake

```diff
@@ -45,15 +45,9 @@ if (WIN32)
         COMPONENT bin)
 endif()
 
-set(PATCH_FILE "${CMAKE_SOURCE_DIR}/thirdparty/date/826_libcpp_17_build_fix.patch")
-set(PC ${Bash_EXECUTABLE}  -c "set -x &&\
-        (\\\"${Patch_EXECUTABLE}\\\" -p1 -R -s -f --dry-run -i \\\"${PATCH_FILE}\\\" || \\\"${Patch_EXECUTABLE}\\\" -p1 -N -i \\\"${PATCH_FILE}\\\")")
-
-
 FetchContent_Declare(date_src
-    URL             https://github.com/HowardHinnant/date/archive/refs/tags/v3.0.1.tar.gz
-    URL_HASH        SHA256=7a390f200f0ccd207e8cff6757e04817c1a0aec3e327b006b7eb451c57ee3538
-    PATCH_COMMAND "${PC}"
+    GIT_REPOSITORY  https://github.com/HowardHinnant/date.git
+    GIT_TAG         1ead6715dec030d340a316c927c877a3c4e5a00c  # master as of 2024-07-25
```

Comment:
I'd prefer to stay on the release tarball version, because fetching tarballs is faster during the build, and github is less likely to rate-limit our builds. Is this feasible?

Reply:
sure; changed to fetching a tarball in 8352c1bc7091a218fb1eaab413ea1a06bc70e35d